hs_core/page_processors.py

```diff
@@ -34,6 +35,16 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
 
     metadata_status = _get_metadata_status(content_model)
 
+    relevant_tools = []
+    for tool in ToolResource.objects.all():
+        if len(tool.metadata.res_types.all())>0:
+            for res_type in tool.metadata.res_types.all():
+                if str(content_model.content_model).lower() in str(res_type.tool_res_type).lower():
+                    tl={'title': tool.title,
+                        'url': tool.metadata.url_bases.first().value+"/?res_id="+content_model.short_id}
```

Comment:
please use string formatting instead of string concatenation, which is very memory allocation heavy. (especially inside a loop)